src/main/java/com/percero/agents/sync/metadata/JpqlQuery.java

```diff
@@ -9,4 +26,78 @@ public void setQuery(String value) {
 			value = value.substring(index + 5);
 		super.setQuery(value);
 	}
+
+	public Query createQuery(Object theObject, String userId, Object[] params,
+			SessionImpl s) throws Exception {
+		String queryString = getQuery();
+		queryString = setQueryParameters(queryString, theObject, userId, params);
+		queryString = setParameterMetadata(queryString, theObject, userId, params, s);
+		Query theQuery = s.createQuery(queryString);
+		
+		return theQuery;
+	}
+	
+	private String setParameterMetadata(String theQuery, Object theObject, String userId, Object[] params, SessionImpl s) {
+		// Get Parameter MetaData.
+		if (params != null && params.length > 0) {
+			if (parameterMetadata == null) {
+				try {
+					parameterMetadata = ((SessionFactoryImpl)s.getSessionFactory()).getQueryPlanCache().getHQLQueryPlan(theQuery, false, (s).getEnabledFilters()).getParameterMetadata();
+				} catch(Exception e) {
+					log.warn("Unable to get ParameterMetaData from Query:\n" + theQuery, e);
+				}
+			}
+			
+			for(Object nextParam : params) {
```

Comment:
This is pretty ugly how nested this is.  You might want to break this down into handler methods for the different branches.